corehq/apps/users/models.py

```diff
@@ -806,13 +814,28 @@ def wrap(cls, data, should_save=False):
             should_save = True
 
         data = cls.migrate_eula(data)
+        data = cls.migrate_device_ids(data)
 
         couch_user = super(CouchUser, cls).wrap(data)
         if should_save:
             couch_user.save()
 
         return couch_user
 
+    @classmethod
+    def migrate_device_ids(cls, data):
+        if (
+            data.get('device_ids') and
+            isinstance(data['device_ids'][0], basestring)
+        ):
+            # device_ids property was only set once, on creation, with a single string, so we only need to worry
+            # about one item in the list
+            data['device_ids'] = [{
+                'device_id': data['device_ids'][0],
```

Comment:
how come you aren't using `DeviceIdLastUsed` here? (alternatively, how come you are using it below if this one was making tests fail)